### PR TITLE
added a shim that can be used to avoid using reflection calls directl…

### LINKED ISSUE #51: Usage is error prone with reflection calls, Proposed strongly typed solution.
Currently, Jar Resolver requires some pretty ugly and unmaintainable reflection calls to get things working. What I have done personally is create a simple class that hides all the reflection and lets you use a strongly typed, easy to read/maintain API to access Jar Resolver.

An example usage of the API is below. It is more clear, less error prone and easier on the eyes than the [SampleDependencies](https://github.com/googlesamples/unity-jar-resolver/blob/0ab833b0b61f5b0d7f14d3a71752ae62282e46cd/sample/Assets/PlayServicesResolver/Editor/SampleDependencies.cs).

```csharp
[InitializeOnLoad]
public class PlayGameServicesDependencies : AssetPostprocessor
{
	static PlayGameServicesDependencies()
	{
		JarResolver.createSupportInstance( typeof( PlayGameServicesDependencies ).Name )
			.dependOn( "com.google.android.gms", "play-services-games", "9+", new string[] { "extra-google-m2repository" } )
			.dependOn( "com.google.android.gms", "play-services-plus", "9+", new string[] { "extra-google-m2repository" } );
	}
}
```

The code for the class is below. If this is something that you would like to have in the repo let me know and I can send over a pull request. If not, feel free to close this issue.

```csharp
namespace Google
{
	public class JarResolver
	{
		public class JarResolverImpl
		{
			object _svcSupport;

			public JarResolverImpl()
			{}


			public JarResolverImpl( object svcSupport )
			{
				_svcSupport = svcSupport;
			}


			public JarResolverImpl DependOn( string group, string artifact, string version, string[] packageIds = null, string[] repositories = null )
			{
				if( _svcSupport != null )
				{
					Google.VersionHandler.InvokeInstanceMethod( _svcSupport, "DependOn",
						new object[] { group, artifact, version },
						namedArgs: new Dictionary<string, object>()
						{
							{ "packageIds", packageIds },
							{ "repositories", repositories }
						});
				}

				return this;
			}
		}


		public static JarResolverImpl CreateSupportInstance( string clientName )
		{
#if !UNITY_ANDROID
			// if we arent on Android default to an empty instance
			return new JarResolverImpl();
#endif

			// bail out with an empty instance if the PlayServicesSupport class isnt available
			var playServicesSupport = Google.VersionHandler.FindClass( "Google.JarResolver", "Google.JarResolver.PlayServicesSupport" );
			if( playServicesSupport == null )
				return new JarResolverImpl();

			// create a live instance of the PlayServicesSupport class and return a live JarResolverImpl wrapping it
			var svcSupport = Google.VersionHandler.InvokeStaticMethod( playServicesSupport, "CreateInstance",
				new object[] {
				clientName,
				EditorPrefs.GetString( "AndroidSdkRoot" ),
				"ProjectSettings"
			} );

			return new JarResolverImpl( svcSupport );
		}
	}
}

```

### PR DESCRIPTION
**Pull request by prime31**
…y, instead sticking with statically typed methods

Fixes #51

### CODE REVIEW

File: sample/Assets/PlayServicesResolver/Editor/Resolver.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+// </copyright>
+
+namespace Google.JarResolver
+{
+	using System;
+	using System.Collections.Generic;
+	using UnityEditor;
+
+	/// <summary>
+    /// Resolver provides an interface to hide the reflection calls required to
```
Comment (stewartmiles) [inline sample/Assets/PlayServicesResolver/Editor/Resolver.cs]: nit: bit off odd indenting going on here, any chance you could fix.  Same below.

File: sample/Assets/PlayServicesResolver/Editor/Resolver.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
+﻿// <copyright file="Resolver.cs" company="Google Inc.">
+// Copyright (C) 2015 Google Inc. All Rights Reserved.
```
Comment (stewartmiles) [inline sample/Assets/PlayServicesResolver/Editor/Resolver.cs]: nit: 2015 --> 2017

**Comment by prime31** [comment]
Damn Visual Studio Mac isnt great at tabs/spaces yet! I just pushed the fixes.

**Comment by stewartmiles** [comment]
Thanks for the update.

**Pull request merged**